core/types/receipt.go

```diff
@@ -411,25 +411,31 @@ func (rs Receipts) DeriveFields(config *params.ChainConfig, hash common.Hash, nu
 		rs[i].Type = txs[i].Type()
 		rs[i].TxHash = txs[i].Hash()
 
+		rs[i].EffectiveGasPrice = txs[i].inner.effectiveGasPrice(new(big.Int), baseFee)
+
 		// block location fields
 		rs[i].BlockHash = hash
 		rs[i].BlockNumber = new(big.Int).SetUint64(number)
 		rs[i].TransactionIndex = uint(i)
 
 		if rs[i].Type != ArbitrumLegacyTxType {
 			// The contract address can be derived from the transaction itself
-			if txs[i].To() == nil && rs[i].ContractAddress == (common.Address{}) {
+			if txs[i].To() == nil {
```

Comment:
We don't want to (and possibly cannot) recalculate ContractAddress if it is already filled (by decodeStoredReceiptRLP),

Reply:
ah, I see this doesn't happen for legacy Tx so I think we always can recalculate it.. question is if we always should

Reply:
I added the check back in, thanks for pointing this out.

Reply:
I get this error from a nitro test when adding the check back in:
```
$ go test -run "TestContractTxDeploy" ./system_tests
...
    contract_tx_test.go:102:  [expected address 0x0000000000000000000000000000123412341234 nonce 0 to deploy to 0x733A0385EB916548E1a2087F038f0dE75043bEF7 but got 0x0000000000000000000000000000000000000000] 
```

Reply:
This check is already inside this if statement, so adding it back into the upper `if` statement would change behavior by causing it to set `ContractAddress` to empty if it was previously set. So I think this code is good as-is.

Reply:
ah yeah, I see that inner if statement was added recently. Makes sense :)